### PR TITLE
[Windows] Fix windows breakage caused by lit.cfg change

### PR DESCRIPTION
**Pull request by artemcm**
Caused by a change in: https://github.com/apple/swift/pull/32903

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

File: test/lit.cfg [hunk 1]
```diff
@@ -1439,24 +1439,28 @@ config.substitutions.append(('%module-target-future', target_future))
...
+path_separator = "/"
+if platform.system() == 'Windows':
+   path_separator = "\"
+
 # Add 'stdlib_dir' as the path to the stdlib resource directory
 stdlib_dir = config.swift_lib_dir + "/swift/"
-if platform.system() == 'Linux':
-    stdlib_dir += config.target_sdk_name + "/" + run_cpu
+if platform.system() == 'Linux' || platform.system() == 'Windows':
+    stdlib_dir += config.target_sdk_name + path_separator + run_cpu
```
Comment (compnerd) [inline test/lit.cfg:1449]: Using `os.path.join` is better IMO, though you could just keep this as is I _think_.

**Comment by artemcm** [comment]
@swift-ci test

File: test/lit.cfg (hunk 1 above)
Comment (artemcm) [reply test/lit.cfg]: A lot better, thanks for the pointer.

**Comment by artemcm** [comment]
@swift-ci Please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by artemcm** [comment]
@swift-ci Please test OS X platform

**Comment by artemcm** [comment]
Looks like the original issue was slightly different, it is referring to an "Unrecognized escape code" inside the json map, not referring to incorrect path separators.

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
I believe that there are two categories of failures, some related to the runtime and others related to the path encoding.

**Comment by artemcm** [comment]
> I believe that there are two categories of failures, some related to the stdlib and others related to the path encoding.

I believe the path encoding failure is caused by the fact that on Linux and macOS, having a comma after the last element of a JSON list is somehow considered to be a valid termination sequence, but on Windows it is not.

**Comment by compnerd** [comment]
Commas after the last element in a list are not valid json: https://www.json.org/json-en.html

**Comment by artemcm** [comment]
Actually it seems that having any line ending with a comma is not valid on Windows? Not necessarily even after the last element in a list...

**Comment by compnerd** [comment]
More likely its line endings ... Windows uses `\r\n` rather than `\n`.

**Comment by compnerd** [comment]
`%ononesupport_module` ... that is going to have `\` in it which is an escape character.   The string you are encoding is the cause of the issue.

**Comment by artemcm** [comment]
It is created as:
`ononesupport_module = os.path.join(stdlib_dir, "SwiftOnoneSupport.swiftmodule")`
and the corresponding line in the lit test that uses the substitution ends with `"`:
`// RUN: echo "\"modulePath\": \"%ononesupport_module\"" >> %/t/inputs/map.json`
Are you referring to the `\` used to escape the `"`?

The error seems to be pointing directly at:
```
T:\swift\test-windows-x86_64\ScanDependencies\Output\explicit-module-map-forwarding-module.swift.tmp/inputs/map.json:10:2: error: Unrecognized escape code

},

 ^
```
Which is separating second and third list elements in the generated JSON list...

**Comment by compnerd** [comment]
The path that is generated from `os.path.join` there will contain `\` which JSON will interpret as an escape sequence even in the quoted string.

**Comment by artemcm** [comment]
I'm confused. Here's the most recent failure log: https://ci-external.swift.org/job/swift-PR-windows/5448/consoleText

The resulting path generated by `os.path.join` is seen in the log in:
```
$ ":" "RUN: at line 29"
$ "echo" ""modulePath": "t:\swift\lib\swift\windows\x86_64\SwiftOnoneSupport.swiftmodule""
```

**Comment by artemcm** [comment]
Ah I see it now in:
```
: 'RUN: at line 29';   echo "\"modulePath\": \"t:\swift\lib\swift\windows\x86_64\SwiftOnoneSupport.swiftmodule\"" >> T:/swift/test-windows-x86_64/ScanDependencies/Output/explicit-module-map-forwarding-module.swift.tmp/inputs/map.json
```

**Comment by compnerd** [comment]
I have a fix, I'll upload it in a second.

**Comment by compnerd** [comment]
#33066 should take care of it

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci please test Windows platform

**Review by compnerd — APPROVED** [review]

**Pull request merged**